src/python/pants/engine/target_test.py

```diff
@@ -37,6 +37,11 @@ def hydrate(self, raw_value: Optional[Iterable[str]], *, address: Address) -> Tu
         return tuple(raw_value)
 
 
+class UnrelatedField(BoolField):
+    alias: ClassVar = "unrelated"
```

Comment:
Can't the type annotation be dropped since the base dictates it?

Reply:
Possibly, I haven't tested that but can test in a followup. With dataclasses, it's sometimes unclear what is a class property vs. instance attribute, so the `ClassVar` is there to disambiguate.

It would remove substantial boilerplate if we could drop this.

Reply:
> so the ClassVar is there to disambiguate.

Makes ~sense.

That said, in standard python anything declared this way must be accessible from the class and dataclasses are no different:
```python
>>> from dataclasses import dataclass
>>> @dataclass
... class Jake:
...   instance_but_doesnt_matter_really_since_default_is_accessibly_from_class: str = "Bob"
... 
>>> Jake.instance_but_doesnt_matter_really_since_default_is_accessibly_from_class
'Bob'
>>> Jake("Jane").instance_but_doesnt_matter_really_since_default_is_accessibly_from_class
'Jane'
>>> 
```